Jellyfin.Server.Implementations/MediaSegments/MediaSegementManager.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Jellyfin.Data.Entities;
+using MediaBrowser.Controller;
+using Microsoft.EntityFrameworkCore;
+
+namespace Jellyfin.Server.Implementations;
+
+/// <summary>
+///     Manages media segments retrival and storage.
+/// </summary>
+public class MediaSegementManager : IMediaSegmentManager
+{
+    private readonly IDbContextFactory<JellyfinDbContext> _dbProvider;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MediaSegementManager"/> class.
+    /// </summary>
+    /// <param name="dbProvider">EFCore Database factory.</param>
+    public MediaSegementManager(IDbContextFactory<JellyfinDbContext> dbProvider)
+    {
+        _dbProvider = dbProvider;
+    }
+
+    /// <inheritdoc />
+    public async Task<MediaSegment> CreateSegmentAsync(MediaSegment mediaSegment)
+    {
+        if (mediaSegment.EndTick < mediaSegment.StartTick)
+        {
+            throw new InvalidOperationException($"A segments {nameof(MediaSegment.EndTick)} cannot be before its {nameof(MediaSegment.StartTick)}");
+        }
```

Comment:
```suggestion
        ArgumentOutOfRangeException.ThrowIfLessThan(mediaSegment.EndTick, mediaSegment.StartTick);
```

Might as well use the new features